packages/medusa/src/api/routes/store/customers/create-address.ts

```diff
@@ -0,0 +1,84 @@
+import { IsOptional, IsString, ValidateNested } from "class-validator"
+import { defaultStoreCustomersFields, defaultStoreCustomersRelations } from "."
+import CustomerService from "../../../../services/customer"
+import { validator } from "../../../../utils/validator"
+
+/**
+ * @oas [post] /customers/me/addresses
+ * operationId: PostCustomersCustomerAddresses
+ * summary: "Add a Shipping Address"
+ * description: "Adds a Shipping Address to a Customer's saved addresses."
+ * x-authenticated: true
+ * requestBody:
+ *   content:
+ *     application/json:
+ *       schema:
+ *         properties:
+ *           address:
+ *             description: "The Address to add to the Customer."
+ *             anyOf:
+ *               - $ref: "#/components/schemas/address"
+ * tags:
+ *   - Customer
+ * responses:
+ *  "200":
+ *    description: "A successful response"
+ *    content:
+ *      application/json:
+ *        schema:
+ *          properties:
+ *            customer:
+ *              $ref: "#/components/schemas/customer"
+ */
+export default async (req, res) => {
+  const id = req.user.customer_id
+
+  const validated = await validator(
+    StorePostCustomersCustomerAddressesReq,
+    req.body
+  )
+
+  const customerService: CustomerService = req.scope.resolve("customerService")
+
+  let customer = await customerService.addAddress(id, validated.address)
+  customer = await customerService.retrieve(id, {
+    relations: defaultStoreCustomersRelations,
+    select: defaultStoreCustomersFields,
+  })
+
+  res.status(200).json({ customer })
+}
+
+export class AddressCreatePayload {
+  @IsString()
+  first_name: string
+  @IsString()
+  last_name: string
+  @IsOptional()
+  @IsString()
+  phone: string
+  @IsOptional()
+  metadata: object
+  @IsOptional()
+  @IsString()
+  company: string
+  @IsString()
+  address_1: string
+  @IsOptional()
+  @IsString()
+  address_2: string
+  @IsString()
+  city: string
+  @IsString()
+  country_code: string
+  @IsOptional()
+  @IsString()
+  province: string
+  @IsString()
+  postal_code: string
+}
```

Comment:
Use common address payload type or if this differs, please move it to the common types, so we can use it other places 👍